dss/events/handlers/sync.py

```diff
@@ -80,12 +84,22 @@ def sync_blob(source_platform, source_key, dest_platform, logger):
         # https://storagetransfer.googleapis.com/$discovery/rest?version=v1
         # doesn't tell me either.
         # gcsts_job = gcsts_conn.api_request("GET", "/" + gcsts_job["name"])
+    elif source_platform == "s3" and dest_platform == "gcs":
+        s3_blob_url = s3.meta.client.generate_presigned_url(
+            ClientMethod='get_object',
+            Params=dict(Bucket=s3_bucket_name, Key=source_key),
+            ExpiresIn=presigned_url_lifetime_seconds
+        )
+        with closing(http.request("GET", s3_blob_url, preload_content=False)) as fh:
+            gcs_bucket = gcs.get_bucket(gcs_bucket_name)
+            gcs_bucket.blob(source_key, chunk_size=1024 * 1024).upload_from_file(fh)
+        logger.info("Completed transfer of {} from {} to {}".format(source_key, s3_bucket_name, gcs_bucket_name))
     elif source_platform == "gcs" and dest_platform == "s3":
-        s3 = boto3.resource("s3")
         gcs_blob = gcs.get_bucket(gcs_bucket_name).blob(source_key)
-        gcs_blob_url = gcs_blob.generate_signed_url(expiration=int(time.time() + presigned_url_lifetime_seconds))
-        http = urllib3.PoolManager(cert_reqs="CERT_REQUIRED")
+        expires_timestamp = int(time.time() + presigned_url_lifetime_seconds)
```

Comment:
strictly speaking, the reading presigned url needs a longer expiration since it got created first.